### PR TITLE
3875 – Create github action to run our tests

### PR DESCRIPTION
**Pull request by vasconsaurus**
## Description

We are migrating Pender tests from Travis to github actions.
To do this we had to figure how we were going to deal with configuration and secrets.
We decided:
- to use the very basic example.config
- get our secrets from ssm into an env file
  - we wrote a script for that
- inject that env file in the container

This job should:
- run tests
- run test coverage, show and upload it
- notify slack (_still needs to be done_)

References: 3875

## How has this been tested?

We ran the action. At first we had some failing tests, but now other than flaky tests it works.

Failing tests:
- one was unrelated (it was making a request to a page that no longer existed)
- some were  some were related to twitter webmocked tests
  - worked locally and on Travis
  - removing the headers fixed it (since they were not necessary, this wasn't an issue)

Flaky tests
- we have some known flaky tests
- and we had some that were some weird flaky webmock ones
  - we should keep our eyes open for these ones 

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: .github/workflows/ci-test-pr.yml
Comment (dmou) [inline .github/workflows/ci-test-pr.yml:6]: make sure to change this to develop!

File: .github/workflows/ci-test-pr.yml
Comment (dmou) [inline .github/workflows/ci-test-pr.yml:19]: for this step and the next four, let's look up those actions on the github actions marketplace and update them to use the most recent version

File: .github/workflows/ci-test-pr.yml
Comment (dmou) [inline .github/workflows/ci-test-pr.yml:99]: looks like your IDE isn't adding newlines to ends of files you're creating btw, you might want to fix that :)

File: .github/workflows/ci-test-pr.yml
Comment (vasconsaurus) [reply .github/workflows/ci-test-pr.yml:6]: @dmou, a question: we still want it to run when we are pushing from our local whatever branch, while we are working on it, right? So could it be something like `on: push` or `on: pull_request`?

File: .github/workflows/ci-test-pr.yml
Comment (vasconsaurus) [reply .github/workflows/ci-test-pr.yml]: fixed it! thank you ❤️

**Comment by vasconsaurus** [comment]
@dmou, some more questions:
- should we remove the travis.yml? 
- should we remove the little travis 'build passing' badge from the readme?
- lastly, sort of unrelated, we have a few code climate links by the side of the travis badge that are all broken. Should we remove those?

Screenshot 2024-02-09 at 11 09 07

**Review by caiosba — APPROVED** [review]
Code changes look good to me, but I defer to @dmou on the other changes.

**Comment by dmou** [comment]
>     * should we remove the travis.yml?

it'll be permanently failing once we merge this in, right? I think we should, we can always revert and restore it if it turns out there's some faulty code and the gha workflow isn't working

>     * should we remove the little travis 'build passing' badge from the readme?

yeah, once we get rid of travis that'll just be confusing so let's take it out

>     * lastly, sort of unrelated, we have a few code climate links by the side of the travis badge that are all broken. Should we remove those?

those don't appear to even resolve to a working url... is there a correct code climate repo link that does work for public view, or does it all require us to log in first? cc @caiosba on that one

**Comment by caiosba** [comment]
@vasconsaurus @dmou , the Code Climate URL for Pender is https://codeclimate.com/github/meedan/pender :)

I agree to remove the Travis badge and file from the repo as part of this merge.

**Comment by vasconsaurus** [comment]
@dmou I made the last changes, can you review it for me when you have some time?

**Review by dmou — APPROVED** [review]

**Pull request merged**
